src/Microsoft.EntityFramework.SqlServer/MicrosoftLocalDbConnectionFactory.cs

```diff
@@ -0,0 +1,119 @@
+        /// </summary>
+        /// <param name="localDbVersion"> The LocalDb version to use. </param>
+        public MicrosoftLocalDbConnectionFactory(string localDbVersion)
+        {
+            Check.NotEmpty(localDbVersion, "localDbVersion");
+
+            _localDbVersion = localDbVersion;
+            _baseConnectionString = @"Integrated Security=True; MultipleActiveResultSets=True;";
+        }
+
+        /// <summary>
+        /// Creates a new instance of the connection factory for the given version of LocalDb.
+        /// For SQL Server 2012 LocalDb use "v11.0".
+        /// For SQL Server 2014 and later LocalDb use "mssqllocaldb".
+        /// </summary>
+        /// <param name="localDbVersion"> The LocalDb version to use. </param>
+        /// <param name="baseConnectionString"> The connection string to use for options to the database other than the 'Initial Catalog', 'Data Source', and 'AttachDbFilename'. The 'Initial Catalog' and 'AttachDbFilename' will be prepended to this string based on the database name when CreateConnection is called. The 'Data Source' will be set based on the LocalDbVersion argument. </param>
+        public MicrosoftLocalDbConnectionFactory(string localDbVersion, string baseConnectionString)
+        {
+            Check.NotEmpty(localDbVersion, "localDbVersion");
+            Check.NotNull(baseConnectionString, "baseConnectionString");
+
+            _localDbVersion = localDbVersion;
+            _baseConnectionString = baseConnectionString;
+        }
+
+        #endregion
+
+        #region Properties
+
+        /// <summary>
+        /// The connection string to use for options to the database other than the 'Initial Catalog',
+        /// 'Data Source', and 'AttachDbFilename'.
+        /// The 'Initial Catalog' and 'AttachDbFilename' will be prepended to this string based on the
+        /// database name when CreateConnection is called.
+        /// The 'Data Source' will be set based on the LocalDbVersion argument.
+        /// The default is 'Integrated Security=True;'.
+        /// </summary>
+        public string BaseConnectionString
+        {
+            get { return _baseConnectionString; }
+        }
+
+        #endregion
+
+        #region IDbConnectionFactory implementation
+
+        /// <summary>
+        /// Creates a connection for SQL Server LocalDb based on the given database name or connection string.
+        /// If the given string contains an '=' character then it is treated as a full connection string,
+        /// otherwise it is treated as a database name only.
+        /// </summary>
+        /// <param name="nameOrConnectionString"> The database name or connection string. </param>
+        /// <returns> An initialized DbConnection. </returns>
+        public DbConnection CreateConnection(string nameOrConnectionString)
+        {
+            Check.NotEmpty(nameOrConnectionString, "nameOrConnectionString");
+
+            var attachDb = " ";
+#if !NETSTANDARD2_1
```

Comment:
We can remove this, right? IIRC, `AttachDbFilename` works now.

Reply:
I think it is more about `|DataDirectory|` - but is if fixed indeed - [I fixed it](https://github.com/dotnet/SqlClient/pull/284) 😄